docs/content/4.sdk/2.getting-started/2.middleware-module.md

```diff
@@ -0,0 +1,471 @@
+# The `middlewareModule`
+
+## Introduction
+
+The `middlewareModule` is an SDK module that ensures communication with the [Server Middleware](/middleware).
+
+## Setup
+
+Depending on which framework you are using, you can get the `middlewareModule` in different ways. However, the setup is always the same.
+
+In Next, `middlewareModule` is available in the `createSdk` function.
+
+```ts [Nuxt]
+// sdk/sdk.config.ts
+import { CreateSdkOptions, createSdk } from "@vue-storefront/next";
+import { UnifiedEndpoints } from "storefront-middleware/types";
+
+const options: CreateSdkOptions = {
+  middleware: {
+    apiUrl: "http://localhost:4000",
+  },
+};
+
+export const { getSdk } = createSdk(
+  options,
+  ({ buildModule, middlewareUrl, middlewareModule, getRequestHeaders }) => ({
+    commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+      apiUrl: middlewareUrl + "/commerce",
+      defaultRequestConfig: {
+        headers: getRequestHeaders(),
+      },
+    }),
+  })
+);
+```
+
+In Nuxt, `middlewareModule` is available in the `defineSdkConfig` function.
+
+```ts [Nuxt]
+// sdk.config.ts
+import { UnifiedEndpoints } from "storefront-middleware/types";
+
+export default defineSdkConfig(
+  ({ buildModule, middlewareUrl, middlewareModule getRequestHeaders }) => ({
+    commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+      apiUrl: middlewareUrl + "/commerce", // SAP Commerce Cloud integration is available at /commerce endpoint
+      defaultRequestConfig: {
+        headers: getRequestHeaders(),
+      },
+    })
+  })
+);
+```
+
+You can also import it directly from the `@vue-storefront/sdk` package.
+
+```ts [Other]
+import { initSDK, buildModule, middlewareModule } from "@vue-storefront/sdk";
+import { UnifiedEndpoints } from "storefront-middleware/types";
+
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+  }),
+});
+```
+
+## Configuration
+
+### Type definition
+
+As the `middlewareModule` could be used to communicate with the Server Middleware and it's not limited to a specific backend, it requires a type definition of the endpoints.
+
+The type definition should be provided as a generic type to the `middlewareModule` function.
+
+```ts
+type Endpoints = {
+  getProduct: ({
+    id: string,
+  }) => Promise<{ id: string; title: string; description: string }>;
+};
+
+const sdk = initSDK({
+  commerce: buildModule(middlewareModule<Endpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+  }),
+});
+
+const product = await sdk.commerce.getProduct({ id: "123" });
+```
+
+In Storefront Starter, you can find the `UnifiedEndpoints` type in `storefront-middleware/types.ts` file. It's a type definition for the endpoints used in the middleware.
+
+```ts
```

Comment:
In the examples we're using the `initSDK` - maybe we should somehow emphasize that for the Next.js/Nuxt you do it differently. Supporting code snippet also for Next.js and Nuxt will be probably an overkill

Reply:
Yeah, I'm going to add a note